### DISCUSSION TITLE
How should Undo history work?

### DISCUSSION THREAD

**Discussion by Ghoulboy78** [Ghoulboy78/World-edit-scarpet#61] [Ideas/Suggestions]
This is an interesting topic, as right now the history is linear, like a stack. If you do A, then B, you must first undo B, then A. And this makes sense, cos if they overwrite the same area, undoing A first, then B would not restore the original area, but maybe that's what you want?

**Comment by Firigion** [comment]
It's an interesting concept. It sounds like hell to implement, but I'm open to ideas.
